pkg/api/util/conditions.go

```diff
@@ -96,17 +96,18 @@ func SetIssuerCondition(i cmapi.GenericIssuer, observedGeneration int64, conditi
 }
 
 // CertificateHasCondition will return true if the given Certificate has a
-// condition matching the provided CertificateCondition.
-// Only the Type and Status field will be used in the comparison, meaning that
-// this function will return 'true' even if the Reason, Message and
+// condition matching the provided CertificateCondition with a ObservedGeneration
+// that is bigger or equal to the ObservedGeneration of the provided CertificateCondition.
+// Only the Type, Status and ObservedGeneration field will be used in the comparison,
+// meaning that this function will return 'true' even if the Reason, Message and
 // LastTransitionTime fields do not match.
 func CertificateHasCondition(crt *cmapi.Certificate, c cmapi.CertificateCondition) bool {
 	if crt == nil {
 		return false
 	}
 	existingConditions := crt.Status.Conditions
 	for _, cond := range existingConditions {
-		if c.Type == cond.Type && c.Status == cond.Status {
+		if c.Type == cond.Type && c.Status == cond.Status && c.ObservedGeneration <= cond.ObservedGeneration {
```

Comment:
What's the reason we are also checking the `ObservedGeneration`?

This does not seem required for the intent of this function since the passed condition normally doesn't have an `ObservedGeneration` set.

If this _is_ required somewhere, can we please add a `CertificateHasConditionWithObservedGeneration` function so we don't accidentally break existing consumers.

Reply:
This function should work for both. Because in case `ObservedGeneration` is not set, it will default to 0 and all `ObservedGeneration` will always be larger than 0.

I'll add a `CertificateHasConditionWithObservedGeneration` function so this is less obscure.

Reply:
From slack;

```
I am very concerned with changing any of our public v1 util functions. These functions are used by third parties and we can't break them.
It could very well be that this doesn't actually break anything, or anyone, but we are changing the meaning of this v1 function.

I'm not too concerned with needing multiple functions, especially in testing. More verbose it better. CertificateHasConditionWithObservedGeneration can even live in /test
```